src/devices/bus/isa/sblaster_lle.cpp

```diff
@@ -0,0 +1,381 @@
+// license:BSD-3-Clause
+// copyright-holders:Katherine Rohl
+/***************************************************************************
+    Sound Blaster LLE device
+    - The "DSP" is an 8051.
+
+    Models:
+    - CT1310 (Sound Blaster 1.0)
+        - In Creative Labs product literature but may not have been released
+        - OPL2, 2x SAA1099, gameport, 22KHz playback/12KHz record
+
+    - CT1320 (Sound Blaster 1.5)
+        - DSP p/n CT1321, version 1.5
+        - Cost-reduced CT1310
+        - OPL2, 2xSAA1099 are now optional, gameport, 22KHz playback/12KHz record
+
+    - CT1350 (Sound Blaster 2.0)
+        - DSP p/n CT1351, version 2.0
+        - OPL2, no SAA1099, 44KHz playback/22KHz record, gameport, auto-initializing DMA
+        - If DSP 2.0x is placed in a CT1320, it becomes functionally equivalent to a CT1350.
+***************************************************************************/
+
+#include "emu.h"
+#include "sblaster_lle.h"
+
+//#define VERBOSE 1
+#include "logmacro.h"
+
+//**************************************************************************
+//  GLOBAL VARIABLES
+//**************************************************************************
+
+#define ym3812_StdClock XTAL(3'579'545)
+#define ymf262_StdClock XTAL(14'318'181)
+#define XTAL_DSP        XTAL(12'000'000)    // XTL1
```

Comment:
Can you use `constexpr` for these rather than macros?  Also, be consistent and use `SCREAMING_SNAKE_CASE` for things that function like constants.